File: ganapathy_pavers/custom/js/quotation.js
frappe.ui.form.on("Quotation", {
    party_name: async function (frm) {
        if (frm.doc.quotation_to == "Lead" && frm.doc.party_name) {
            await frappe.db.get_value("Lead", frm.doc.party_name, "type").then(res => {
                frm.set_value("order_type", res["message"]["type"])
            })
        }
    },
    before_validate: function (frm) {
        if (!frm.doc.order_type) {
            frm.trigger("party_name")
        }
    },
    onload: function (frm) {
        if (frm.is_new() && !frm.doc.order_type) {
            frm.trigger("party_name")
        }
    },
    refresh: function (frm) {
        frm.set_query("supervisor_name", function () { 
            return {
                filters: {
                    designation: "Supervisor"
                }
            }
        })
    },
})

File: ganapathy_pavers/custom/py/quotation.py
import frappe

def update_lead_on_save(self, event = None):
    if self.quotation_to == "Lead" and self.party_name:
        if frappe.get_value("Lead",self.party_name,"status") == "Lead":
            frappe.get_doc("Lead", self.party_name).set_status(update=True, status="Quotation")